piet-wgsl/shader/fine.wgsl

```diff
@@ -281,7 +281,10 @@ fn main(
     for (var i = 0u; i < PIXELS_PER_THREAD; i += 1u) {
         let coords = xy_uint + vec2(i, 0u);
         if coords.x < config.target_width && coords.y < config.target_height {
-            textureStore(output, vec2<i32>(coords), rgba[i]);
+            let fg = rgba[i];
+            let a_inv = 1.0 / (fg.a + 1e-6);
```

Comment:
Is there any firm reasoning behind the use of specifically $10^{-6}$ here?
It probably needs a comment explaining the choice of magic number.

Reply:
This exists simply to avoid a divide by zero without adding a branch. Happy to add a comment noting that, similar to the one in blend.wgsl.

Reply:
This is to avoid NaN. The idea is that it's a number small enough that it won't affect visual rendering, but not so small you might get overflow. 1e-6 is a pretty big "epsilon" as far as f32's go, but this code was originally ported from GLSL which had "mediump" and we do want to support f16 (yet another permutation, sigh).

One thought that occurs to me is that min() might be better than +, as it would leave pretty much all nonzero values untouched.

Reply:
I think you mean max() rather than min()? I'll make that change (in blend as well)

Reply:
Yes max, duh.

Reply:
I don't think I made myself clear here; I knew why this addition was done; it's quite clear it's to avoid divide by zero.
My question was more about why $10^-6$ as opposed to say $10^-7$ or $10^-5$. It's fine if it's basically arbitrary (or in this case, other historical reasons?)